### PR TITLE
add "sensor_pose_estimation based control" flag in launch file.

### PR DESCRIPTION
**Pull request by tongtybj**
Add the arg "use_ground_truth" in launch file. 
When the  arg "use ground truth" is true, no pose estimation(egomotion) is processed, and the control is based on the ground truth value. 

On the other hand, when the arg is false, the pose estimation begins to integrate sensors (c.f. imu, gps, baro(hieght)) to generate a combined 6-dof state(odmetry). The control is then based on the odometry from the pose estimation. 

The gps and baro sensor model plugin is provide by either hector_quadcopter_gazebo_plugin or hector_gazebo_plugin which already contain the noise(based on Gussian). This issue is related to #63 and #64

### CODE REVIEW

**Comment by tongtybj** [comment]
also add ground hit check function which is mentioned in #62 .
Right now I just check the relative height of drone based on the ground. But I think real contact detection is better. @k-okada how to detect the contact or collision between two models in gazebo?

File: jsk_mbzirc_common/src/mbzirc_gazebo_truck_plugin.cpp [hunk 1]
```diff
@@ -103,6 +107,25 @@ void GazeboTruck::Update()
...
+  if(!drone_model) return;
+  if(!takeoff_flag_)
+    {//TODO: check contact with model "arena" is better
+      if(landing_height_ < 0) landing_height_  = drone_model->GetWorldPose().pos.z;
+      if(drone_model->GetWorldPose().pos.z > landing_height_ + 0.05) //0.05m is a margin
+        takeoff_flag_ = true;
+    }
+  else
+    {//TODO: check contact with model "arena" is better
+      if(drone_model->GetWorldPose().pos.z < landing_height_ + 0.05)
```
Comment (tongtybj) [inline jsk_mbzirc_common/src/mbzirc_gazebo_truck_plugin.cpp]: @k-okada  A simple check of hitting ground by checking the relative height towards the ground.

**Comment by k-okada** [comment]
as far as I understand, this is "truck" plugin, so he does not understand where the drone is. to get ground truth, we'd use/update hector_quadcopter_gazebo_plugin or we can use p3d plugin (
http://gazebosim.org/tutorials?tut=ros_gzplugins#P3D(3DPositionInterfaceforGroundTruth)
)

>  how to detect the contact or collision between two models in gazebo?

i think https://github.com/start-jsk/jsk_mbzirc/blob/master/jsk_mbzirc_common/src/mbzirc_gazebo_truck_plugin.cpp#L159 is good starting point

**Comment by tongtybj** [comment]
@k-okada Crushing  after hitting ground is a special rule for task1. In task3, drones may always hit to ground to pick up the object. So, hardcoding  in  hector_quadcopter_gazebo_plugin may not be a good choice. Adding  an additional plugin like "task1 plugin" is my suggestion. Actually, the "truck" plugin contains lot of rules about task1 beyond the truck movement. So I put code into the truck plugin.
 And in p3d plugin, should I use  "bumper" plugin to realize the contact?

**Comment by k-okada** [comment]
in task3, the drone will not collide with ground, but boxes.

--
◉ Kei Okada

**Comment by k-okada** [comment]
by the why, this features is an optional, not mandatory , just to check if
someone invent unrealistic strategies.

--
◉ Kei Okada

**Comment by tongtybj** [comment]
Right now I also added the gimbal module for sensor in this branch.

**Comment by furushchev** [comment]
@tongtybj For keeping higher quality of your great contribution, you might be better to:
- split this Pull Request to some no-more-decomposable units
  - e.g. `add mbzirc_gazebo_gimbal_plugin`, `add XXXplugin`, `update rviz for gimbal visualization`...
- `mbzirc_gazebo_drone_additional_plugin` is ambiguous. It will be better to describe what is actually `additional` feature.

**Comment by tongtybj** [comment]
Separate into #82~#84. Closed this.

**Pull request closed without merging**